homeassistant/helpers/frame.py

```diff
@@ -144,24 +145,72 @@ def report(
     If error_if_integration is True, raise instead of log if an integration is found
     when unwinding the stack frame.
     """
+    core_behavior = ReportBehavior.ERROR if error_if_core else ReportBehavior.LOG
+    core_integration_behavior = (
+        ReportBehavior.ERROR if error_if_integration else ReportBehavior.LOG
+    )
+    custom_integration_behavior = core_integration_behavior
+
+    if log_custom_component_only:
+        if core_behavior == ReportBehavior.LOG:
+            core_behavior = ReportBehavior.IGNORE
+        if core_integration_behavior == ReportBehavior.LOG:
+            core_integration_behavior = ReportBehavior.IGNORE
+
+    report_usage(
+        what,
+        core_behavior=core_behavior,
+        core_integration_behavior=core_integration_behavior,
+        custom_integration_behavior=custom_integration_behavior,
+        exclude_integrations=exclude_integrations,
+        level=level,
+    )
+
+
+class ReportBehavior(Enum):
+    """Enum for behavior on code usage."""
+
+    IGNORE = 1
+    """Ignore the code usage."""
+    LOG = 2
+    """Log the code usage."""
+    ERROR = 3
+    """Raise an error on code usage."""
+
```

Comment:
Since these don't need to be numbers, I'd go with `auto()` so its clear the value is not to be used externally

Reply:
Done